Formula/l/localai.rb

```diff
@@ -0,0 +1,69 @@
+class Localai < Formula
+  include Language::Python::Virtualenv
+
+  desc ":robot: The free, Open Source OpenAI alternative"
+  homepage "https://localai.io"
+  url "https://github.com/mudler/LocalAI/archive/refs/tags/v2.18.1.tar.gz"
+  sha256 "54dd442498263657b8953a767db0a31ad86591130095a42d84ecbbd8c795d880"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "go" => :build
+  depends_on "python-setuptools" => :build
+  depends_on xcode: :build
+
+  depends_on "abseil"
+  depends_on "grpc"
+  depends_on "protobuf"
+  depends_on "protoc-gen-go"
+  depends_on "protoc-gen-go-grpc"
+  depends_on "python@3.12"
+  depends_on "wget"
+
+  resource "grpcio-tools" do
+    url "https://files.pythonhosted.org/packages/9f/30/cd31c3a04814eb880d5e78cea768240c92fb5adaa158814c2b166356a0c6/grpcio_tools-1.64.0.tar.gz"
+    sha256 "fa4c47897a0ddb78204456d002923294724e1b7fc87f0745528727383c2260ad"
+  end
+
+  def install
+    ENV["PYTHON"] = python3 = which("python3.12")
+
+    venv = virtualenv_create(libexec, python3)
+    venv.pip_install(resources, build_isolation: false)
+
+    # We exclude gpt4all backend because it is causing an issue between
+    # CMAKE_OSX_ARCHITECTURES and the CXX compiler targets architectures. Check
```

Comment:
do we have issue tracking this?

Reply:
@chenrui333 issue in which project you mean?

Reply:
I guess whatever project that we can followup on this :) I was thinking localai

Reply:
It's not an issue of LocalAI itself, the gpt4all backend can only be built one arch at a time, so passing the multi-arch will always break. I don't think this should block this PR from being merged though. I'm already working for a cleaner way to exclude backends in LocalAI, which I can use once the next release is out, but it would be very helpful if we could have the formula already merged

Reply:
@SMillerDev good question, I'm not sure why the runner has `CMAKE_OSX_ARCHITECTURES` set to "arm64;x86_64" which is why this error pops up

```
CMake Error at /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeDetermineCompilerABI.cmake:129 (message):
    The CXX compiler targets architectures:
  
      "arm64"
  
    but CMAKE_OSX_ARCHITECTURES is
  
      "arm64;x86_64"
 ```
 
 sorry not a macOS expert here, so cannot really tell if this is expected in general or just on the runners

Reply:
I tried in different ways to set `CMAKE_OSX_ARCHITECTURES` just the currently running arch, but that didn't work. I think that is related to how the shell out in ruby does the passing of the environment variables

Reply:
It's not supposed to be an environment variable, how about setting `-DCMAKE_OSX_ARCHITECTURES=#{Hardware::CPU.arch}` as a CMake arg?

Reply:
@mauromorales can we try out the cmake arg route?